### PR TITLE
fix: align readme with current mteb

### PR DESCRIPTION
**Pull request by Samoed**
## Checklist

- [X] Run tests locally to make sure nothing is broken using `make test`. 
- [X] Run the formatter to format the code using `make lint`. 

The current README is a bit misleading and causing issues for newcomers. See #1490 and #1491 for reference.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,8 @@ model_name = "average_word_embeddings_komninos"
 # model_name = "sentence-transformers/all-MiniLM-L6-v2"
 
 model = SentenceTransformer(model_name)
+# or directly from mteb:
+model = mteb.get_model(model_name)
```
Comment (KennethEnevoldsen) [inline README.md:54]: Should we just recommend this one always? 

and maybe rephrase to:
```suggestion
model = mteb.get_model(model_name) # if the model is not implemented in MTEB it will be eq. to SentenceTransformer(model_name)
```

File: README.md [hunk 2]
```diff
@@ -220,9 +222,13 @@ Note that the public leaderboard uses the test splits for all datasets except MS
...
 
 ```python
+import mteb
 from mteb.encoder_interface import PromptType
+from mteb.models.wrapper import Wrapper
+import numpy as np
+
 
-class CustomModel:
+class CustomModel(Wrapper):
```
Comment (KennethEnevoldsen) [inline README.md:231]: Why does it need to inherit from wrapper? I would instead inherit from the Encoder protocol

Comment (Samoed) [reply README.md:231]: MTEB class will wrap models that not inherit wrapper class https://github.com/embeddings-benchmark/mteb/blob/3ff38ec043f6ebc4e434c083cf96dc823439f8a3/mteb/evaluation/MTEB.py#L366-L367

File: README.md (hunk 1 above)
Comment (Samoed) [reply README.md:54]: Yes, but I'm not sure removing it is the best approach. I can make changes to whatever you think is better

File: README.md (hunk 2 above)
Comment (KennethEnevoldsen) [reply README.md:231]: Hmm, maybe it should just wrap SentenceTransformers?

Comment (Samoed) [reply README.md:231]: It can be like this. You changed it in mieb https://github.com/embeddings-benchmark/mteb/blob/fab0b82a5c2680ab4e4823641955cdc7b9bb9891/mteb/evaluation/MTEB.py#L356-L357 I can change there too

Comment (KennethEnevoldsen) [reply README.md:231]: Hmm then it will also be merged into v2.0.0 in which case we should probably just update the readme there (I plan to merge it during December)

Comment (Samoed) [reply README.md:231]: So, for now I think this PR can be merged, and I will update readme in the 2.0 branch

**Comment by Samoed** [comment]
I made model check as in `mieb` branch

**Review by KennethEnevoldsen — APPROVED** [review]
Yep - pull from main should do

**Comment by KennethEnevoldsen** [comment]
This seems ready to merge (for future reference feel free to just merge once you have gotten an approved review)
